lib/libzfs/libzfs_util.c

```diff
@@ -596,27 +596,47 @@ zfs_strdup(libzfs_handle_t *hdl, const char *str)
  * Convert a number to an appropriately human-readable output.
  */
 void
-zfs_nicenum(uint64_t num, char *buf, size_t buflen)
+__zfs_nicenum(uint64_t num, char *buf, size_t buflen,
```

Comment:
I'm not convinced this approach is better than simply leaving zfs_nicenum() as is, and adding zfs_nicetime() and zfs_nicenum10(). It seems like a lot of conditional code to try to share a single interface. Conditional code should be avoided when possible.